proto/redpanda/api/dataplane/v1/pipeline.proto

```diff
@@ -353,6 +353,28 @@ service PipelineService {
     };
   }
 
+  // GetComponentList returns a JSON containing lists of allowed components based on component type filter.
+  rpc GetComponentList(GetComponentListRequest) returns (GetComponentListResponse) {
```

Comment:
[nit] I think `ListComponents` would be better? Usually we reserve `Get` prefix for obtaining 1 item